src/main/java/jenkins/branch/BuildRetentionBranchProperty.java

```diff
@@ -40,22 +40,20 @@ public P project(@NonNull final P project) {
                 // ==========
                 // The lesser evil hack is to set the field by reflection, so we try that first
 
-                Boolean success = AccessController.doPrivileged(new PrivilegedAction<Boolean>() {
-                    public Boolean run() {
+                Boolean success = AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
+                    try {
+                        Field logRotator = Job.class.getDeclaredField("logRotator");
```

Comment:
Hmm, this is going to break in Java 17, need to introduce at least a `Beta` core API…